libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,280 @@
+import { AfterViewInit, ChangeDetectorRef, ChangeDetectionStrategy, Component, EventEmitter } from '@angular/core';
+import { Input, NgZone, Optional, Output, Self, ViewChild, ViewEncapsulation } from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { CheckboxComponent as CoreCheckboxComponent } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+import { FormFieldControl } from '../form-control';
+
+/** Change event object emitted by Platform Checkbox. */
+export class PlatformCheckboxChange {
+    /** The source Checkbox of the event. */
+    source: CheckboxComponent;
+    /** The new `checked` value of the checkbox. */
+    checked: boolean;
+}
+
+/**
+ * Checkbox implementation based on the
+ * https://github.com/SAP/fundamental-ngx/wiki/Platform:-Checkbox-Component-Technical-Design
+ * documents.
+ *
+ * This implementation behaves like implementation in PrimeNg and Material checkbox implementation.
+ * Some part of code/idea has been taken from above mentioned and has been implemented to work with platform form.
+ * primeng: https://primefaces.org/primeng/showcase/#/checkbox
+ */
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [{ provide: FormFieldControl, useExisting: CheckboxComponent, multi: true }]
+})
+export class CheckboxComponent extends BaseInput implements AfterViewInit {
+    /** set to true if binary checkbox */
+    @Input()
```

Comment:
I am more or less ok with this version I would work on more on the Selection Model and remove all the code and replace it with Selection Model that exists in the Material/cdk. it will make things much cleaner